pkg/virt-operator/resource/apply/generations.go

```diff
@@ -75,53 +68,42 @@ func SetGeneration(generations *[]k6tv1.GenerationStatus, actual runtime.Object)
 		return
 	}
 
-	operatorGenerations := toOperatorGenerations(*generations)
 	meta := actual.(v1.Object)
-
-	resourcemerge.SetGeneration(&operatorGenerations, operatorsv1.GenerationStatus{
+	setGeneration(generations, k6tv1.GenerationStatus{
 		Group:          group,
 		Resource:       resource,
 		Namespace:      meta.GetNamespace(),
 		Name:           meta.GetName(),
 		LastGeneration: meta.GetGeneration(),
 	})
-
-	newGenerations := toAPIGenerations(operatorGenerations)
-	*generations = newGenerations
 }
 
-func toOperatorGeneration(generation k6tv1.GenerationStatus) operatorsv1.GenerationStatus {
-	return operatorsv1.GenerationStatus{
-		Group:          generation.Group,
-		Resource:       generation.Resource,
-		Namespace:      generation.Namespace,
-		Name:           generation.Name,
-		LastGeneration: generation.LastGeneration,
-		Hash:           generation.Hash,
-	}
-}
+func generationFor(generations []k6tv1.GenerationStatus, resource schema.GroupResource, namespace, name string) *k6tv1.GenerationStatus {
+	for i := range generations {
+		curr := &generations[i]
+		if curr.Namespace == namespace &&
+			curr.Name == name &&
+			curr.Group == resource.Group &&
+			curr.Resource == resource.Resource {
 
-func toAPIGeneration(generation operatorsv1.GenerationStatus) k6tv1.GenerationStatus {
-	return k6tv1.GenerationStatus{
-		Group:          generation.Group,
-		Resource:       generation.Resource,
-		Namespace:      generation.Namespace,
-		Name:           generation.Name,
-		LastGeneration: generation.LastGeneration,
-		Hash:           generation.Hash,
+			return curr
+		}
 	}
+
+	return nil
 }
 
-func toOperatorGenerations(generations []k6tv1.GenerationStatus) (operatorGenerations []operatorsv1.GenerationStatus) {
-	for _, generation := range generations {
-		operatorGenerations = append(operatorGenerations, toOperatorGeneration(generation))
+func setGeneration(generations *[]k6tv1.GenerationStatus, newGeneration k6tv1.GenerationStatus) {
```

Comment:
Would it make sense to merge `SetGeneration` and `setGeneration`? Is there a point in keeping them separate?

Reply:
done